app/workers/after_user_create_worker.rb

```diff
@@ -51,13 +51,16 @@ def associate_ownerships(user, email)
   def associate_membership_invites(user, email, without_confirm: false)
     memberships = Membership.unclaimed.where(invited_email: email)
     return false unless memberships.any?
-    memberships.pluck(:id).each_with_index do |m_id, index|
-      if index == 0 # We want to do the first one inline so we can redirect the user to the org page
-        ProcessMembershipWorker.new.perform(m_id, user.id)
-      else
-        ProcessMembershipWorker.perform_async(m_id, user.id)
-      end
+
+    first, *rest = memberships.pluck(:id)
```

Comment:
Was this added to Rails, or is it a new ruby feature? it's nifty regardless.

Reply:
Just a refactor made while I was debugging — the first/rest array-deconstruction pattern is just FP-flavored Ruby.

I intended it to be temporary but left it in because I thought it expressed intent more clearly (process the first synchronously, the rest asynchronously). Happy to revert if it seems too terse though.

Reply:
No, I like it a lot!